code/modules/html_interface/voting/voting.dm

```diff
@@ -239,6 +239,16 @@ var/global/datum/controller/vote/vote = new()
 	if(mode)
 		if(config.vote_no_dead && usr.stat == DEAD && !usr.client.holder)
 			return 0
+		if(mode == "map")
+			var/mob/M = usr
+			if(istype(M, /mob/new_player))
+				to_chat(usr, "<span class='warning'>Only players that have joined the round may vote for the next map.</span>")
+				return 0
+			if(isobserver(M))
+				var/mob/dead/observer/O = M
+				if(O.started_as_observer)
```

Comment:
Not as familiar with vote code as I should be, would this prevent admins from voting? Since this and the block above it don't check `holder`

Reply:
haha thank you force push, you are easier than squashing